lib/utils/oauth/src/index.js

```diff
@@ -278,8 +283,82 @@ const cache = (authServer, id, secret, scopes) => {
   return wrapper;
 };
 
+// Decode basic token
+const decodeBasicToken = (auth) => new Buffer(auth.split(' ')[1], 'base64')
+  .toString().split(':');
+
+// Get token from auth server, expect to have the required scope
+const basicAuth = (authServer, auth, cb) => {
+  if (!secured())
+    return cb(undefined, true);
+  tokenEndpoint(authServer, (err, endpoint) => {
+    if (err) {
+      edebug('Error getting OAuth bearer access token, %o', err);
+      debug('Error getting OAuth bearer access token, %o', err);
+      return cb({
+        statusCode: 401,
+        header: {
+          'WWW-Authenticate': 'Basic realm="cf", error="invalid_token",' +
+            ' error_description="malformed ' + err + ' "'
+        }
+      }, false);
+    }
+    let decoded;
+    try {
+      decoded = decodeBasicToken(auth);
+    }
+    catch (exception) {
+      edebug('Exception when decoding basic token, %o', exception);
+      debug('Exception when decoding basic token, %o', exception);
+  
+      return cb({
+        statusCode: 401,
+        header: {
+          'WWW-Authenticate': 'Bearer realm="cf", error="invalid_token",' +
+            ' error_description="malformed"'
+        }
+      }, false);
+    }
+    const id = decoded[0];
+    const secret = decoded[1];
+    // Give access only to user with the right username and password
+    const scope = id === username && secret === password ?
```

Comment:
Not sure why you need to check for this username and password here in the client library as UAA is responsible to authenticate the user for you.